### PR TITLE
Add "fsadmin pathConf list" command

### PR DESCRIPTION
**Pull request by cheng-chang**
This command lists paths that have path level configurations.

### CODE REVIEW

File: shell/src/main/java/alluxio/cli/fsadmin/command/PathConfCommand.java [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+      return mClass;
+    }
+  }
+
+  /** A map from sub-commands' names to the definition. */
+  private static final Map<String, SubCommand> SUB_COMMAND_MAP = new HashMap<>();
+
+  static {
+    for (SubCommand cmd : SubCommand.values()) {
+      SUB_COMMAND_MAP.put(cmd.getName(), cmd);
```
Comment (calvinjia) [inline shell/src/main/java/alluxio/cli/fsadmin/command/PathConfCommand.java]: Do we need this map, could we use `valueOf` and some kind of case normalization?

File: shell/src/main/java/alluxio/cli/fsadmin/command/PathConfCommand.java [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+/**
+ * Manages path level configuration.
+ */
+public final class PathConfCommand extends AbstractFsAdminCommand {
+  private enum SubCommand {
+    LIST("list", ListCommand.class),
+    ;
+
+    private final String mName;
+    private final Class<? extends Command> mClass;
```
Comment (aaudiber) [inline shell/src/main/java/alluxio/cli/fsadmin/command/PathConfCommand.java]: consider storing a `Map<String, BiFunction<Context, AlluxioConfiguration, ? extends Command>>` instead. Then we don't need reflection at all

**Comment by cheng-chang** [comment]
@calvinjia this command only lists the paths, there will be another command showing configurations for a path.

File: shell/src/main/java/alluxio/cli/fsadmin/command/PathConfCommand.java (hunk 1 above)
Comment (cheng-chang) [reply shell/src/main/java/alluxio/cli/fsadmin/command/PathConfCommand.java]: yes, possible, but then `LiSt` will also be recognized as the `list` command.

**Comment by cheng-chang** [comment]
@calvinjia @aaudiber PTAL

File: shell/src/main/java/alluxio/cli/fsadmin/command/PathConfCommand.java (hunk 1 above)
Comment (calvinjia) [reply shell/src/main/java/alluxio/cli/fsadmin/command/PathConfCommand.java]: Got it, thanks.

**Comment by calvinjia** [comment]
@cheng-chang If someone calls this command when they have thousands of paths configured, it will output all of them right? I am thinking of providing a prefix so they can see the ones under a subtree.

File: shell/src/main/java/alluxio/cli/fsadmin/command/PathConfCommand.java [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+    StringBuilder usage = new StringBuilder(getCommandName());
+    for (String cmd : SUB_COMMANDS.keySet()) {
+      usage.append(" [").append(cmd).append("]");
+    }
+    return usage.toString();
+  }
+
+  @Override
+  public String getDescription() {
+    return "Manage path level configuration, see sub-commands' descriptions for more details.";
```
Comment (aaudiber) [inline shell/src/main/java/alluxio/cli/fsadmin/command/PathConfCommand.java]: is there a way for users to discover the names of the subcommands?

File: shell/src/main/java/alluxio/cli/fsadmin/pathconf/ListCommand.java [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+import alluxio.wire.Configuration;
+
+import org.apache.commons.cli.CommandLine;
+
+import java.io.IOException;
+
+/**
+ * Lists paths that have path level configuration.
+ */
+public final class ListCommand extends AbstractFsAdminCommand {
```
Comment (aaudiber) [inline shell/src/main/java/alluxio/cli/fsadmin/pathconf/ListCommand.java]: does this need to extend `AbstractFsAdminCommand`? Usually things extending that can be called directly like `fsadmin list`, not via subcommand. See `ReportCommand` and its subcommands for example

File: shell/src/main/java/alluxio/cli/fsadmin/command/PathConfCommand.java (hunk 3 above)
Comment (cheng-chang) [reply shell/src/main/java/alluxio/cli/fsadmin/command/PathConfCommand.java]: `getUsage` lists the names of the subcommands.

File: shell/src/main/java/alluxio/cli/fsadmin/pathconf/ListCommand.java (hunk 4 above)
Comment (cheng-chang) [reply shell/src/main/java/alluxio/cli/fsadmin/pathconf/ListCommand.java]: by extending `AbstractFsAdminCommand`, the subcommand's interface and implementation can have the same structure as other commands. From a user's perspective, a subcommand is also a fsadmin command.

**Comment by cheng-chang** [comment]
@calvinjia in the design, we expected only a small number of paths are configured with path defaults. So I think we're good at least for now.

**Comment by aaudiber** [comment]
@cheng-chang can we add a test for the new command?

**Comment by calvinjia** [comment]
@cheng-chang LGTM, please add testing

**Comment by cheng-chang** [comment]
@aaudiber I'll add a test after https://github.com/Alluxio/alluxio/pull/8640 is merged, it depends on functionalities there.

**Comment by cheng-chang** [comment]
@aaudiber @calvinjia PTAL

File: tests/src/test/java/alluxio/client/cli/fsadmin/pathconf/ListCommandIntegrationTest.java [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+ */
+public class ListCommandIntegrationTest extends PathConfTest {
+  @Test
+  public void list() {
+    int ret = mFsAdminShell.run("pathConf", "list");
+    Assert.assertEquals(0, ret);
+    String output = mOutput.toString();
+    String expected = DIR1 + "\n" + DIR2 + "\n";
+    Assert.assertEquals(expected, output);
+  }
```
Comment (calvinjia) [inline tests/src/test/java/alluxio/client/cli/fsadmin/pathconf/ListCommandIntegrationTest.java]: The construction of the FS tree is in another class which makes this hard to follow. Also, should we test the case where there are some directories which do not have path level config?

Comment (cheng-chang) [reply tests/src/test/java/alluxio/client/cli/fsadmin/pathconf/ListCommandIntegrationTest.java]: @calvinjia I made it more straightforward, PTAL.

**Comment by calvinjia** [comment]
alluxio-bot, merge this please.
